File: py-test/dimacs_test.py
#! /usr/bin/env python3

### @file dimacs_test.py
### @brief
### @author Yusuke Matsunaga (松永 裕介)
###
### Copyright (C) 2018 Yusuke Matsunaga
### All rights reserved.

import unittest
from pym_combopt import UdGraph

# UdGraph のテスト用クラス
class UdGraphTest(unittest.TestCase) :

    def runTest(self) :
        self.test_dimacs()


    def test_dimacs(self) :
        filename = 'anna.col'

        graph = UdGraph.read_dimacs(filename)

        # graph.node_num == 138
        # graph.edge_num == 986

        ofilename = '??'
        graph.write_dimacs(ofilename)

        # filename == ofilename


if __name__ == '__main__' :

    unittiest.main()

File: c++-test/udgraph/dimacs_test.cc

/// @file dimacs_test.cc
/// @brief dimacs_test の実装ファイル
/// @author Yusuke Matsunaga (松永 裕介)
///
/// Copyright (C) 2018 Yusuke Matsunaga
/// All rights reserved.


#include "gtest/gtest.h"
#include "ym/UdGraph.h"


BEGIN_NAMESPACE_YM

TEST(UdGraphDimacsTest, read_dimacs)
{
  string filename = string(TESTDATA_DIR) + string("/anna.col");
  ifstream s(filename);
  ASSERT_COND( static_cast<bool>(s) );

  UdGraph graph = UdGraph::read_dimacs(s);

  ASSERT_EQ( 138, graph.node_num() );
  ASSERT_EQ( 986, graph.edge_num() );

  ostringstream obuf;
  graph.write_dimacs(obuf);

  istringstream s1(obuf.str());
  UdGraph graph2 = UdGraph::read_dimacs(s1);
  ASSERT_EQ( 130, graph2.node_num() );

  ostringstream obuf2;
  graph2.write_dimacs(obuf2);

  EXPECT_EQ( obuf2.str(), obuf.str() );
}

END_NAMESPACE_YM
